doc/userguide/rules/intro.rst

```diff
@@ -248,10 +251,54 @@ Now, let's say we have a rule with the following header::
 Only the traffic from the client to the server will be matched by this rule,
 as the direction specifies that we do not want to evaluate the response packet.
 
+Now,if we have a rule with the following header::
+
+    alert tcp 1.2.3.4 any <> 5.6.7.8 80
+
+Suricata will duplicate it and use the same rule with headers in both directions :
+
+    alert tcp 1.2.3.4 any -> 5.6.7.8 80
+    alert tcp 5.6.7.8 80 -> 1.2.3.4 any
+
 .. warning::
 
    There is no 'reverse' style direction, i.e. there is no ``<-``.
 
+Bidirectional rules
+~~~~~~~~~~~~~~~~~~~
+
+Here is an example of a bidirectional rule:
+
+.. container:: example-rule
+
+    alert http any any :example-rule-emphasis:`=>` 5.6.7.8 80 (msg:"matching both uri and status"; sid: 1; http.uri; content: "/download"; http.stat_code; content: "200";)
+
+It will match on flows to 5.6.7.8 and port 80.
+And it will match on a full transaction, using both the uri from the request,
+and the stat_code from the response.
+As such, it will match only when Suricata got both request and response.
+
+Bidirectional rules can use direction-ambiguous keywords, by first using
+``bidir.toclient`` or ``bidir.toserver`` keywords.
```

Comment:
not a fan of this additional keyword, can we use the flow keyword for it? Also, I guess I'm not even sure what they are used for? Can you explain it?

Reply:
> Also, I guess I'm not even sure what they are used for? 

This is important !
These are helper to the rule parser to know for ambiguous keywords such as `http.connection` (which can happen on both directions) which direction to use.

Like `http.uri; content: "/download"; http.stat_code; content: "200"; bidir.toclient; http.connection; content: "eep";`
means that we should use the `http.connection; content: "eep";` sent to the client
Do you have a better idea or a better naming ?

Reply:
Ok, I see. I think an alternative would be to only allow `=>` rules for non-ambiguous keywords, so keywords that have explicit directionality. E.g. instead of `... http ... => ... file.data; ...` force ppl to use `... http ... => ... http.request_body; ...`.

Reply:
Sure, v1 was like that, but you told me to remove limitations because signature writers would not want them :-p

Reply:
Maybe reusing `flow: toserver` instead of introducing `bidir.toserver` is a better idea...